### PR TITLE
PUBDEV-8212: cleanup + improve efficiency of job polling in AutoML

### PR DESCRIPTION
**Pull request by sebhrusen**
https://h2oai.atlassian.net/browse/PUBDEV-8212

As job polling in AutoML also means polling the event_log (that's the part creating all the noise), I also standardized it to use the shared log levels now available with the "new" logger.

### CODE REVIEW

File: h2o-py/h2o/automl/_estimator.py [hunk 1]
```diff
@@ -73,19 +73,29 @@ class H2OAutoML(H2OAutoMLBaseMixin, Keyed):
...
-    The current version of AutoML trains and cross-validates the following algorithms (in the following order):
-    three pre-specified XGBoost GBM (Gradient Boosting Machine) models,
-    a fixed grid of GLMs,
-    a default Random Forest (DRF),
-    five pre-specified H2O GBMs,
-    a near-default Deep Neural Net,
-    an Extremely Randomized Forest (XRT),
-    a random grid of XGBoost GBMs,
-    a random grid of H2O GBMs,
-    and a random grid of Deep Neural Nets.
```
Comment (sebhrusen) [inline h2o-py/h2o/automl/_estimator.py:85]: @ledell I updated this description as it is outdated now. Feel free to rephrase.

File: h2o-r/h2o-package/R/automl.R [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
 #' The Automatic Machine Learning (AutoML) function automates the supervised machine learning model training process.
-#' The current version of AutoML trains and cross-validates the following algorithms:
-#' three pre-specified XGBoost GBM (Gradient Boosting Machine) models, a fixed grid of GLMs,
-#' a default Random Forest (DRF), five pre-specified H2O GBMs, a near-default Deep Neural Net,
-#' an Extremely Randomized Forest (XRT), a random grid of XGBoost GBMs, a random grid of H2O GBMs,
-#' and a random grid of Deep Neural Nets. In some cases, there will not be enough time to complete all the algorithms,
-#' so some may be missing from the leaderboard. AutoML trains several Stacked Ensemble models during the run.
-#' Two kinds of Stacked Ensemble models are trained one of all available models, and one of only the best models of each kind.
-#' Note that Stacked Ensemble models are trained only if there isn't another stacked ensemble with the same base models.
-#'
```
Comment (sebhrusen) [inline h2o-r/h2o-package/R/automl.R:12]: @ledell same as for Python description.
Given the R constraints in this main description (stops after the first blank line apparently), I moved the general legacy `@details` section to the top, and put the internals description into the new `@details` section, as this way I can use markdown there.

**Review by tomasfryda — COMMENTED** [review]
There are some failing tests that seem to be related to the change. Otherwise, it looks good to me. Thank you @sebhrusen !

File: h2o-py/h2o/automl/_estimator.py [hunk 3]
```diff
@@ -73,19 +73,29 @@ class H2OAutoML(H2OAutoMLBaseMixin, Keyed):
...
+    It trains several models, cross-validated by default, by using the following available algorithms:
+    
+    - XGBoost
+    - GBM (Gradient Boosting Machine)
+    - GLM (Generalized Linear Model)
+    - DRF (Distributed Random Forest)
+    - XRT (eXtremely Randomized Trees)
+    - DeepLearning
```
Comment (ledell) [inline h2o-py/h2o/automl/_estimator.py:83]:
```suggestion
    - DeepLearning
```
```suggestion
    - DeepLearning (Fully Connected Deep Neural Network)
```

File: h2o-r/h2o-package/R/automl.R [hunk 4]
```diff
@@ -69,11 +63,31 @@
...
-#' @details AutoML finds the best model, given a training frame and response, and returns an H2OAutoML object,
-#'          which contains a leaderboard of all the models that were trained in the process, ranked by a default model performance metric.
 #' @return An \linkS4class{H2OAutoML} object.
+#' @details AutoML trains several models, cross-validated by default, by using the following available algorithms:
+#'  * XGBoost
+#'  * GBM (Gradient Boosting Machine)
+#'  * GLM (Generalized Linear Model)
+#'  * DRF (Distributed Random Forest)
+#'  * XRT (eXtremely Randomized Trees)
+#'  * DeepLearning
```
Comment (ledell) [inline h2o-r/h2o-package/R/automl.R:75]:
```suggestion
#'  * DeepLearning (Fully Connected Deep Neural Network)
```

**Review by ledell — APPROVED** [review]
This looks good, thanks!

**Pull request merged**